packages/server/src/services/digester.ts

```diff
@@ -1,47 +1,114 @@
 import {cEvent} from "./consensus";
+import {Storage} from "./storage";
+import {Transaction} from "../types/transaction";
+import {State} from "../types/state";
 
 export class Digester<T> {
 
+    /**
+     * Class constructor.
+     *
+     * @param storage the interface for storage
+     */
+    constructor(private storage: Storage) {
+
+    }
+
     /**
      * Digests all the events that have reached consensus
      *
      * @param events All events that are not yet processed and have reached consensus
      *
      */
-    public digest(events: cEvent<T>[]): void {
+    public async digest(events: cEvent<Transaction[]>[]): Promise<void> {
+        for (let i = 0; i < events.length; i++) {
+            const event = events[i];
+
+            await this.handleTransactions(event.data)
```

Comment:
Ik zie dat je voor elk event `handleTransactions` aanroept maar op het moment heeft één event maximaal één transactie, toch?

Reply:
ik dacht dat het uiteindelijk de bedoeling was om meerdere mee te kunnen sturen. Nu is de mogelijkheid er wel maar je kan ook gewoon 1 mee geven.

Reply:
Jaa dat kunnen we wel doen maar dan moet je de andere code ook even aanpassen. Als we hier nu een transactie heen gooien geeft die op een type error of gooit die een error omdat niet over een type Transaction heen kan loopen

Reply:
ligt er aan zien we het nog gebeuren dat we meerdere transacties in een event gaan stoppen anders bouw ik dat om dan kunnen we de rest met rust laten. Als we het wel van plan zijn zal ik het even aanpassen.

Reply:
Jaa wat mij betreft doen we dat nu gewoon. Eén kan altijd maar meer is dan ook mogelijk.